cmd/net.go

```diff
@@ -0,0 +1,110 @@
+/*
+ * Minio Cloud Storage, (C) 2017 Minio, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package cmd
+
+import (
+	"fmt"
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/minio/minio-go/pkg/set"
+)
+
+// splitHostPort is wrapper to net.SplitHostPort(), but ignores "missing port in address".
+// If port is missing, default port "9000" is returned.
+func splitHostPort(hostPort string) (host, port string, err error) {
+	host, port, err = net.SplitHostPort(hostPort)
+	// Ignore "missing port in address" error as we use default port.
+	if err != nil && strings.Contains(err.Error(), "missing port in address") {
+		host = hostPort
+		port = "9000"
+		err = nil
+	}
+
+	p, err := strconv.Atoi(port)
+	if err != nil {
+		return "", "", fmt.Errorf("Invalid port number")
+	} else if p < 1 {
+		return "", "", fmt.Errorf("Port number should be greater than zero")
+	}
+
+	return host, port, err
+}
+
+func mustSplitHostPort(hostPort string) (host, port string) {
+	host, port, err := splitHostPort(hostPort)
+	if err != nil {
+		panic(err)
+	}
+
+	return host, port
+}
+
+// mustGetLocalIP4 returns IPv4 addresses of local host.  It panics on error.
+func mustGetLocalIP4() (ipList set.StringSet) {
```

Comment:
It looks like we are only supporting IP v4 for the moment, can't we add support of IP v6 at the same time?